node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodePrioritizer.java

```diff
@@ -320,4 +318,27 @@ private boolean isDocker() {
                 .filter(n -> n.hostname().equals(node.parentHostname().orElse(" NOT A NODE")))
                 .findAny();
     }
+
+    private static int compareForRelocation(Node a, Node b) {
+        // Choose smallest node
+        int capacity = ResourceCapacity.of(a).compare(ResourceCapacity.of(b));
+        if (capacity != 0) return capacity;
+
+        // Choose unallocated over allocated (this case is when we have ready docker nodes)
+        if (!a.allocation().isPresent() && b.allocation().isPresent()) return -1;
+        if (a.allocation().isPresent() && !b.allocation().isPresent()) return 1;
+
+        // Choose container over content nodes
+        if (a.allocation().isPresent() && a.allocation().isPresent()) {
```

Comment:
One of these should be **b**`.allocation().isPresent()`

Reply:
I must say! Very nice spotted. This looks bad for the unit test I wrote for this too.

Reply:
Ah.. I see. due to the condition above, the && condition is not necessary.  I'll change it as it is more readable instead of removing it.

Reply:
Props! Not often you see this kind of errors spotted in reviews.